### PR TITLE
Adding Petstore Client R Package

### PR DESCRIPTION
**Pull request by ramnov**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: master for non-breaking changes and `3.0.0` branch for breaking (non-backward compatible) changes.

### Description of the PR
Petstore Client R Package
(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by wing328** [comment]
@ramnov thanks for the PR. I'll use it to create the R client generator.

File: samples/client/petstore/R/LICENSE [hunk 1]
```diff
@@ -0,0 +1,2 @@
+YEAR: 2017
+COPYRIGHT HOLDER: Ramkumar Chandrasekaran
```
Comment (wing328) [inline samples/client/petstore/R/LICENSE]: @ramnov FYI. The auto-generated code is unlicensed so we won't generate this file.

Comment (ramnov) [inline samples/client/petstore/R/LICENSE]: Ok. This License File is not required. Will send another PR to remove it
